libs/product/src/models/configurable-product.ts

```diff
@@ -1,21 +1,24 @@
 import { DaffSortable } from '@daffodil/core';
 
 import { DaffProduct } from './product';
+import { DaffProductImage } from './product-image';
 
 export interface DaffConfigurableProduct extends DaffProduct {
-	configurableOptions: DaffConfigurableProductOption[];
-	variants?: DaffConfigurableProductVariant[];
+	configurableAttributes: DaffConfigurableProductAttribute[];
+	variants: DaffConfigurableProductVariant[];
 }
 
-export interface DaffConfigurableProductOption extends DaffSortable {
+export interface DaffConfigurableProductAttribute extends DaffSortable {
 	code:	string;
 	label: string;
 	values:	DaffConfigurableProductOptionValue[];
 }
 
 export interface DaffConfigurableProductVariant {
 	appliedAttributes: DaffProductVariantAttributesDictionary;
-	product: DaffProduct;
```

Comment:
I think these should be subkeys of `product`, I like the original interface better, I think its more explicit.

Reply:
Won't it be confusing, since it isn't really a product?

Reply:
I think it might make more sense to treat the entire `DaffConfigurableProductVariant` as the product, since that's kind of what it is. It's just that it has applied attributes.